components/api/src/main/java/com/hotels/styx/api/Url.java

```diff
@@ -119,7 +116,7 @@ public boolean isSecure() {
      */
     public boolean isFullyQualified() {
         Optional<String> host = host();
-        return host.isPresent() && !isNullOrEmpty(host.get());
+        return host.isPresent() && !host.get().isEmpty();
```

Comment:
This is a change in behaviour since a `null` host would have returned `false` before. Is this expected?

Reply:
`isPresent()` is checking for null, so I took the opportunity to simplify other check.

Reply:
Yes I think Optional<> cannot store null values.